### PR TITLE
Elemental analysis crash with bad input

### LINKED ISSUE #32309: Elemental analysis crash
I can cause a crash in elemental analysis by trying to change the peaks file.

Open elemental analysis
Go to Edit and then load the attached file (it has a non-existent element)
Mantd error reporter will appear

I think we can stop this by adding a check that the "element" is in the expected elements list

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**

The elemental analysis (EA) GUI would crash if the reference data file contained a non-existent element. This PR fixes that.

**To test:**

Download a copy of the attached file.
Open EA
Go to `Edit` and select `change peak data file`
Select the attached file.
There should only be two elements (H and Ag) enabled

Open the file and edit it by changing H to something thats not an element (e.g. Hi) - save the file
Go to edit ... and load the file
This time only Ag should be enabled

Open the file and edit it by changing Ag to something thats also not an element (e.g. Age) - save the file
Go to edit ... and load the file
This time a warning should appear and the GUI will be set to the default (same as start up)

Fixes #32309  

In release notes

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v6.2.0/muon.rst [hunk 1]
```diff
@@ -102,6 +102,11 @@ Improvements
 ############
 - Updated :ref:`LoadElementalAnalysisData <algm-LoadElementalAnalysisData>` algorithm to include Poisson errors for the counts data.
 
+Bug Fixes
```
Comment (sf1919) [inline docs/source/release/v6.2.0/muon.rst:105]: Please can you change this to Bugfixes? This is in keeping with the rest of the release notes.

File: scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py [hunk 2]
```diff
@@ -388,22 +396,26 @@ def select_data_file(self):
...
                 self.ptable.select_element(element)
             else:
                 self._remove_element_lines(element)
         self._update_checked_data()
 
+    def _reset_data_file_warning_and_action(self):
+        message_box.warning(
+            'The file does not contain correctly formatted data, resetting to default data file.'
+            'See "https://docs.mantidproject.org/nightly/interfaces/'
+            'Muon%20Elemental%20Analysis.html" for more information.')
```
Comment (sf1919) [inline scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py:415]: The URL is incorrect. It should be https://docs.mantidproject.org/nightly/interfaces/muon/Muon%20Elemental%20Analysis.html (there is a folder for muon interfaces within the interfaces folder)

**Review by sf1919 — CHANGES_REQUESTED** [review]
I have tested the PR using the test script and everything works as it should - no crashes occurred. However looking at the code there are some minor changes that need to be made.

File: docs/source/release/v6.2.0/muon.rst (hunk 1 above)
Comment (AnthonyLim23) [reply docs/source/release/v6.2.0/muon.rst:105]:
```suggestion
Bugfixes
```

File: scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py (hunk 2 above)
Comment (AnthonyLim23) [reply scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py:415]:
```suggestion
            'See "https://docs.mantidproject.org/nightly/interfaces/muon/'
            'Muon%20Elemental%20Analysis.html" for more information.')
```

File: docs/source/release/v6.2.0/muon.rst [hunk 3]
```diff
@@ -102,6 +102,11 @@ Improvements
...
+#########
```
Comment (AnthonyLim23) [inline docs/source/release/v6.2.0/muon.rst:106]:
```suggestion
########
```

**Review by sf1919 — APPROVED** [review]
I'm happy that the changes I requested have been implemented.

**Pull request merged**